optout/src/main/java/gov/cms/ab2d/optout/OptOutProcessorImpl.java

```diff
@@ -1,69 +1,78 @@
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Value;
+import org.apache.commons.io.IOUtils;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
-import software.amazon.awssdk.core.exception.SdkClientException;
-import software.amazon.awssdk.core.exception.SdkServiceException;
-import software.amazon.awssdk.regions.Region;
-import software.amazon.awssdk.services.s3.S3Client;
-import software.amazon.awssdk.services.s3.model.GetObjectRequest;
 
 import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.Optional;
+
 
 @Slf4j
 @Component
+@RequiredArgsConstructor
 public class OptOutProcessorImpl implements OptOutProcessor {
 
-    @Value("${s3.region}")
-    private String s3Region;
+    private final S3Gateway s3Gateway;
+    private final ConsentRepository consentRepository;
+    private final ConsentConverterService  consentConverterService;
 
-    @Value("${s3.bucket}")
-    private String s3Bucket;
 
-    @Value("${s3.filename}")
-    private String s3Filename;
 
     @Override
     @Transactional
     public void process() {
 
-        //set region
-        final Region region = Region.of(s3Region);
-
+        final InputStreamReader inputStreamReader = s3Gateway.getS3Object();
 
-        // build S3 client
-        final S3Client s3Client =  S3Client.builder().region(region).build();
+        try (var bufferedReader = new BufferedReader(inputStreamReader)) {
+            importConsentRecords(bufferedReader);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
 
+    }
 
-        // build GetObjectRequest
-        final GetObjectRequest getObjectRequest = GetObjectRequest.builder()
-                .bucket(s3Bucket)
-                .key(s3Filename)
-                .build();
 
+    private void importConsentRecords(BufferedReader bufferedReader) {
+        var iterator = IOUtils.lineIterator(bufferedReader);
 
+        int linesRead = 0;
+        int insertedRowCount = 0;
+        int skippedRowCount = 0;
+        while (iterator.hasNext()) {
+            ++linesRead;
 
-        long linesReadCount = 0;
-        try {
-            var responseInputStream = s3Client.getObject(getObjectRequest);
+            try {
+                final String line = iterator.nextLine();
 
-            var bufferedReader = new BufferedReader(new InputStreamReader(responseInputStream));
-            linesReadCount = bufferedReader.lines().count();
+                Optional<Consent> optConsent = consentConverterService.convert(line, linesRead);
+                if (optConsent.isPresent()) {
+                    consentRepository.save(optConsent.get());
+                    ++insertedRowCount;
+                }
+            } catch (Exception e) {
+                log.error("Invalid opt out record {}", linesRead, e);
```

Comment:
Logging the line number here is sufficient.

Reply:
done